dali/base/dafdesc.cpp

```diff
@@ -390,6 +390,7 @@ struct CClusterInfo: implements IClusterInfo, public CInterface
     Linked<IGroup> group;
     StringAttr name; // group name
     ClusterPartDiskMapSpec mspec;
+    bool foreignGroup = false;
```

Comment:
How does foreignGroup ever get set?  It looks like it is only set when deserializing from a property tree.  Is that initially in extracting the information from the foreign dali when IFDSF_FOREIGN_GROUP is set?
When it is coming from a serialized foreign dali it then has the foreign flag set?

Reply:
yes, when it gets a foreign file, it gets a property tree, and calls deserializeFileDescriptorTree with flags=IFDSF_FOREIGN_GROUP, which constructs CClusterInfo via a property tree, passing through the IFDSF_FOREIGN_GROUP.
That ctor, sets foreignGroup and it is preserved through serialization (when CDistributedFile::getFileDescriptor serializes) through @foreign in serializeTree.

>When it is coming from a serialized foreign dali it then has the foreign flag set?

so it's not coming from the remote side, the:
```
    Owned<IFileDescriptor> fdesc = deserializeFileDescriptorTree(tree,&queryNamedGroupStore(),IFDSF_FOREIGN_GROUP);
```

in CDistributedFileDirectory::getFile( on a foreign will propagate it through during deserialization.